--- libtirpc-1.0.2/tirpc/rpc/rpcent.h.0010~	2017-11-07 10:57:15.269002291 +0100
+++ libtirpc-1.0.2/tirpc/rpc/rpcent.h	2017-11-07 10:58:37.694954696 +0100
@@ -48,8 +48,9 @@
 extern "C" {
 #endif
 
-/* These are defined in /usr/include/rpc/netdb.h */
-#if !defined(__GLIBC__) || defined(__UCLIBC__)
+/* These are defined in /usr/include/rpc/netdb.h, unless we are using
+ * a C library without RPC support. */
+#if !defined(__GLIBC__) || (defined(__UCLIBC__) && !defined(__UCLIBC_HAS_RPC__))
 struct rpcent {
 	char	*r_name;	/* name of server for this rpc program */
 	char	**r_aliases;	/* alias list */
