--- libtirpc-0.2.4/Makefile.am.0011~	2014-07-02 19:14:00.986996853 +0200
+++ libtirpc-0.2.4/Makefile.am	2014-07-02 19:14:43.697412094 +0200
@@ -1,6 +1,10 @@
 SUBDIRS = src man doc rpcgen
 ACLOCAL_AMFLAGS = -I m4
 
+
+GENFILES = \
+			 tirpc/rpc/rpcb_prot.h
+
 noinst_HEADERS	       = tirpc/reentrant.h \
 			 tirpc/getpeereid.h \
 			 tirpc/libc_private.h \
@@ -20,7 +24,6 @@ nobase_include_HEADERS = tirpc/netconfig
 			 tirpc/rpc/rpcent.h \
 			 tirpc/rpc/rpc_com.h \
 			 tirpc/rpc/rpcb_prot.x \
-			 tirpc/rpc/rpcb_prot.h \
 			 tirpc/rpc/rpcb_clnt.h \
 			 tirpc/rpc/raw.h \
 			 tirpc/rpc/pmap_rmt.h \
@@ -38,8 +41,21 @@ nobase_include_HEADERS = tirpc/netconfig
 			 tirpc/rpc/auth_gss.h \
 			 tirpc/rpc/auth_des.h
     
+nobase_nodist_include_HEADERS = $(GENFILES)
+
+BUILT_SOURCES = $(GENFILES)
+
 pkgconfigdir=$(libdir)/pkgconfig
 pkgconfig_DATA = libtirpc.pc
 
-CLEANFILES	       = cscope.* *~
+$(GENFILES): %.h: %.x $(top_builddir)/rpcgen/rpcgen
+	mkdir -p $(dir $@)
+	$(top_builddir)/rpcgen/rpcgen -h -o $@ $<
+
+$(top_builddir)/rpcgen/rpcgen: force
+	cd rpcgen && $(MAKE)
+
+force:
+
+CLEANFILES	       = $(GENFILES) |cscope.* *~
 DISTCLEANFILES	       = Makefile.in libtirpc*.tar.gz
--- libtirpc-0.2.4/src/Makefile.am.0011~	2014-07-02 19:14:01.016997145 +0200
+++ libtirpc-0.2.4/src/Makefile.am	2014-07-02 19:14:01.020330511 +0200
@@ -8,7 +8,7 @@
 noinst_HEADERS = rpc_com.h
 
 AM_CPPFLAGS = -I$(top_srcdir)/tirpc -DPORTMAP -DINET6 \
-		-D_GNU_SOURCE -Wall -pipe  
+		-D_GNU_SOURCE -Wall -pipe -I$(top_builddir)/tirpc
 
 lib_LTLIBRARIES = libtirpc.la
 
