--- libtirpc-1.0.1/src/rpc_com.h.omv~	2015-11-11 19:59:54.621046175 +0100
+++ libtirpc-1.0.1/src/rpc_com.h	2015-11-11 20:00:06.879969915 +0100
@@ -61,8 +61,8 @@ void __xprt_unregister_unlocked(SVCXPRT
 void __xprt_set_raddr(SVCXPRT *, const struct sockaddr_storage *);
 
 
-SVCXPRT **__svc_xports;
-int __svc_maxrec;
+extern SVCXPRT **__svc_xports;
+extern int __svc_maxrec;
 
 #ifdef __cplusplus
 }
--- libtirpc-1.0.1/src/svc.c.omv~	2015-11-11 20:00:10.408947997 +0100
+++ libtirpc-1.0.1/src/svc.c	2015-11-11 20:00:14.771920920 +0100
@@ -800,3 +800,6 @@ rpc_control (int what, void *arg)
     }
   return FALSE;
 }
+
+SVCXPRT **__svc_xports;
+int __svc_maxrec;
